packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -1075,25 +1155,26 @@ export class GarbageCollector implements IGarbageCollector {
 			}
 		}
 
-		if (!this.configs.shouldRunSweep) {
-			return sweepPhaseStats;
-		}
-
 		// If sweep is enabled, the counts from the mark phase stats do not include nodes that have been
-		// deleted in previous runs. Add the deleted node counts to life time stats.
+		// deleted in previous runs. So, add the deleted node counts to life time stats.
 		sweepPhaseStats.lifetimeNodeCount += sweepPhaseStats.deletedNodeCount;
 		sweepPhaseStats.lifetimeDataStoreCount += sweepPhaseStats.deletedDataStoreCount;
 		sweepPhaseStats.lifetimeAttachmentBlobCount += sweepPhaseStats.deletedAttachmentBlobCount;
 
-		// The node deleted in current run are counted twice - once in allDeletedNodes and again in
-		// markPhaseStats. So, remove them from the life time stats.
-		for (const nodeId of currentDeletedNodes) {
-			sweepPhaseStats.lifetimeNodeCount--;
+		if (this.configs.sweepEnabled) {
+			return sweepPhaseStats;
+		}
+
+		// If sweep is not enabled, the current sweep ready node stats should be added to deleted stats since this
```

Comment:
I guess I'm still skeptical about this - seems like it will make this event hard to interpret in general.  We can discuss later, it's fine to keep it this way for now.

Reply:
Let's just keep this for now. We can revisit it later.